src/drive/components/IntentButton.jsx

```diff
@@ -0,0 +1,68 @@
+import styles from '../styles/intentbutton'
+
+/* global cozy */
+import React from 'react'
+import classNames from 'classnames'
+
+import Modal from 'cozy-ui/react/Modal'
+
+class IntentButton extends React.Component {
+  state = {
+    modalIsOpen: false
+  }
+
+  openModal() {
+    this.setState({
+      modalIsOpen: true
+    })
+  }
+
+  closeModal = () => {
+    this.setState({
+      modalIsOpen: false
+    })
+  }
+
+  render() {
+    const { className, data, intent, children } = this.props
+    const { modalIsOpen } = this.state
+    return (
+      <span className={classNames(styles['intentButton'])}>
+        {modalIsOpen && (
+          <Modal secondaryAction={() => this.closeModal()}>
+            <Intent data={data} intent={intent} closeModal={this.closeModal} />
+          </Modal>
+        )}
+        <button
+          className={className}
+          onClick={() => this.openModal(data, intent)}
+        >
+          {children}
+        </button>
+      </span>
+    )
+  }
+}
+
+class Intent extends React.Component {
+  componentDidMount() {
+    const { data, intent, closeModal } = this.props
+    cozy.client.intents
+      .create(intent.action, intent.docType, {
+        ...data,
+        exposeIntentFrameRemoval: true
+      })
+      .start(document.getElementById('intentViewer'))
+      .then(() => {
```

Comment:
If you put `exposeIntentFrameRemoval` to `true`, you can get `removeIntentFrame` which allow to remove the intent HTML node after calling `closeModal()` (and avoid blank page inside the modal)